reflex/compiler/compiler.py

```diff
@@ -189,18 +194,74 @@ def _compile_root_stylesheet(stylesheets: list[str]) -> str:
         if get_config().tailwind is not None
         else []
     )
+
+    sass_compile = None
     for stylesheet in stylesheets:
         if not utils.is_valid_url(stylesheet):
             # check if stylesheet provided exists.
-            stylesheet_full_path = (
-                Path.cwd() / constants.Dirs.APP_ASSETS / stylesheet.strip("/")
-            )
+            assets_app_path = Path.cwd() / constants.Dirs.APP_ASSETS
+            stylesheet_full_path = assets_app_path / stylesheet.strip("/")
+
             if not stylesheet_full_path.exists():
                 raise FileNotFoundError(
                     f"The stylesheet file {stylesheet_full_path} does not exist."
                 )
-            stylesheet = f"../{constants.Dirs.PUBLIC}/{stylesheet.strip('/')}"
+
+            if stylesheet_full_path.is_dir():
+                # NOTE: this can create an infinite loop, for example:
+                # assets/
+                #   | dir_a/
+                #   |   | dir_c/ (symlink to "assets/dir_a")
+                #   | dir_b/
+                # so to avoid the infinite loop, we don't include symbolic links
+                stylesheets += [
+                    str(p.relative_to(assets_app_path))
+                    for p in stylesheet_full_path.iterdir()
+                    if not (p.is_symlink() and p.is_dir())
+                ]
+                continue
+
+            if (
+                stylesheet_full_path.suffix[1:].lower()
+                in constants.Reflex.STYLESHEETS_SUPPORTED
+            ):
+                target = (
+                    Path.cwd()
+                    / constants.Dirs.WEB
+                    / constants.Dirs.STYLES
+                    / RE_SASS_SCSS_EXT.sub(".css", str(stylesheet)).strip("/")
+                )
+                target.parent.mkdir(parents=True, exist_ok=True)
+
+                if stylesheet_full_path.suffix == ".css":
+                    path_ops.cp(src=stylesheet_full_path, dest=target, overwrite=True)
+                else:
+                    try:
+                        from sass import compile as sass_compile
+
+                        target.write_text(
+                            data=sass_compile(
+                                filename=str(stylesheet_full_path),
+                                output_style="compressed",
+                            ),
+                            encoding="utf8",
+                        )
+                    except ImportError:
+                        sass_compile = None
+            else:
+                raise FileNotFoundError(
+                    f'The stylesheet file "{stylesheet_full_path}" is not a valid file.'
+                )
+
+            stylesheet = f"./{str(stylesheet).replace(stylesheet_full_path.suffix, ".css").strip('/')}"
+
         sheets.append(stylesheet) if stylesheet not in sheets else None
+
+    if sass_compile is None:
```

Comment:
You should name this something along the lines `failed_to_import_sass` and let it be a boolean variable.